### PR TITLE
example-log: add support for --log-size

### PR DESCRIPTION
**Pull request by cjhoward92**
Updated the `log` example to use the `--log-size` parameter.

### CODE REVIEW

File: examples/log.c [hunk 1]
```diff
@@ -63,7 +63,7 @@ struct log_options {
 static int parse_options(
 	struct log_state *s, struct log_options *opt, int argc, char **argv);
 static void print_time(const git_time *intime, const char *prefix);
-static void print_commit(git_commit *commit);
+static void print_commit(git_commit *commit, int show_log_size);
```
Comment (mikeando) [inline examples/log.c]: It would feel more logical to me to pass a pointer to the  `log_options` struct here.

File: examples/log.c [hunk 2]
```diff
@@ -49,7 +49,7 @@ static int add_revision(struct log_state *s, const char *revstr);
 
 /** log_options holds other command line options that affect log output */
 struct log_options {
-	int show_diff;
+	int show_diff, show_log_size;
```
Comment (mikeando) [inline examples/log.c]: Maybe a new line for this?

File: examples/log.c [hunk 3]
```diff
@@ -347,6 +347,11 @@ static void print_commit(git_commit *commit)
 	git_oid_tostr(buf, sizeof(buf), git_commit_id(commit));
 	printf("commit %s\n", buf);
 
+	if (show_log_size) {
+		printf("log size %d", (int)strlen(git_commit_message(commit)));
+		printf("\n");
```
Comment (mikeando) [inline examples/log.c]: Why not just one printf call?

~~~
printf("log size %d\n", (int)strlen(git_commit_message(commit)));
~~~

**Comment by cjhoward92** [comment]
@mikeando Thanks for the review and feedback! Hopefully my changes are up to snuff!

**Comment by carlosmn** [comment]
The format matches git and the implementation looks straightforward. Thanks!

**Pull request merged**